microsoft-edge/webview2/concepts/user-data-folder.md

```diff
@@ -1,65 +1,637 @@
+
+For example code, see the WinUI 2 (UWP) `.cs` file, at [WebView2Samples repo > webview2_sample_uwp](https://github.com/MicrosoftEdge/WebView2Samples/tree/master/SampleApps/webview2_sample_uwp).
+
+
+
+<!-- ====================================================================== -->
+<!-- custom UDF location -->
+
+# [WinUI 3](#tab/winui3)
+
+<!--
+**Should you use default or custom UDF location?**
+-->
+On this platform, use the default UDF location.
+
+For example code, see the WinUI 3 `.cs` file, at [WebView2Samples repo > WebView2_WinUI3_Sample](https://github.com/MicrosoftEdge/WebView2Samples/tree/master/SampleApps/WebView2_WinUI3_Sample).
+
+
+**Where browser data gets stored within the UDF:**
+
+After creation of the session and UDF, browser data from your WebView2 control is stored in a subfolder of `userDataFolder`.
+
+
+---
+
+<!-- end of "custom UDF location" tab-set -->
+
+
+<!-- ====================================================================== -->
+## Retrieving the UDF location
+
+To find out what the user data folder location was set to, use the `UserDataFolder` property.  This read-only property returns the user data folder location for the WebView2 app.
+
+Reasons you might want to read the UDF location:
+
+*  If you want to clear browsing data from the UDF folder, such as at the end of a session.
+
+*  If you want to delete the user data folder.
+
+
+<!-- ====================================================================== -->
+<!-- retrieving UDF location -->
+
+# [Win32](#tab/win32)
+
+Use the Win32 [ICoreWebView2Environment7.get_UserDataFolder property getter](https://docs.microsoft.com/microsoft-edge/webview2/reference/win32/icorewebview2environment7#get_userdatafolder).  That API Reference page contains example code showing how to read the `UserDataFolder` property.
+
+**Sample code:**
+
+```cpp
+auto environment7 = m_webViewEnvironment.try_query<ICoreWebView2Environment7>();
+CHECK_FEATURE_RETURN(environment7);
+wil::unique_cotaskmem_string userDataFolder;
+environment7->get_UserDataFolder(&userDataFolder);
+```
+
+For examples of reading the `UserDataFolder` property, see the Win32 samples in the [WebView2Samples repo](https://github.com/MicrosoftEdge/WebView2Samples).
+
+
+<!-- ====================================================================== -->
+<!-- retrieving UDF location -->
+
+# [.NET (WPF/WinForms)](#tab/dotnet)
+
+Use the .NET [CoreWebView2Environment.UserDataFolder Property](https://docs.microsoft.com/dotnet/api/microsoft.web.webview2.core.corewebview2environment.userdatafolder).
+
+<!-- dev: add example code to https://docs.microsoft.com/dotnet/api/microsoft.web.webview2.core.corewebview2environment.userdatafolder showing how to read the `UserDataFolder` property, copy that from the below code block: -->
+
+**Sample code:**
+
+```csharp
+// todo: Dev to provide now - doesn't exist in sample repo at this time
```

Comment:
@DavidShoe - pls provide code to retrieve UDF location on .NET (WPF/WinForms)

Reply:
We need a bug to track this work, it is going to require updating samples.